test/pkg/environment/common/expectations.go

```diff
@@ -146,9 +160,12 @@ func (env *Environment) ExpectCreatedOrUpdated(objects ...client.Object) {
 
 func (env *Environment) ExpectSettings() (res []corev1.EnvVar) {
 	GinkgoHelper()
-
+	// we cannot run tests for now on karpenter if the controller isn't in cluster
+	if !env.InClusterController() {
+		return
+	}
```

Comment:
I would strongly prefer gating _calls_ to these expectations, rather than checking within expectations themselves. Silently skipping the check that calling context expects (pun) is not a good pattern - convenient as it is.